spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KafkaStreamsBinderSupportAutoConfiguration.java

```diff
@@ -282,6 +286,18 @@ else if (configProperties
 				Collectors.toMap((e) -> String.valueOf(e.getKey()), Map.Entry::getValue));
 	}
 
+	private void addPrefix(Properties properties, Map<String, Object> mergedConsProdConfig, String prefix) {
+		Map<String, Object> mergedConfigs = new HashMap<>();
+		for (String key : mergedConsProdConfig.keySet()) {
+			if (!key.startsWith(prefix)) {
+				mergedConfigs.put(prefix + key, mergedConsProdConfig.get(key));
+			}
```

Comment:
```suggestion
			mergedConfig.put(key.startsWith(prefix) ? key : prefix + key, mergedConsProdConfig.get(key));
```

Also need to add if already prefixed.